### PR TITLE
Make registerMarker's cursor offset optional

### PR DESCRIPTION
**Pull request by childrentime**
#3611

### CODE REVIEW

File: src/browser/public/Terminal.ts [hunk 1]
```diff
@@ -166,7 +166,7 @@ export class Terminal implements ITerminalApi {
     this._checkProposedApi();
     this._core.deregisterCharacterJoiner(joinerId);
   }
-  public registerMarker(cursorYOffset: number): IMarker | undefined {
```
Comment (Tyriar) [inline src/browser/public/Terminal.ts:169]: Let's add `: number` back here (personal preference)

File: src/headless/public/Terminal.ts [hunk 2]
```diff
@@ -136,7 +136,7 @@ export class Terminal implements ITerminalApi {
     this._verifyIntegers(columns, rows);
     this._core.resize(columns, rows);
   }
-  public registerMarker(cursorYOffset: number): IMarker | undefined {
```
Comment (Tyriar) [inline src/headless/public/Terminal.ts:139]: Same

**Review by Tyriar — CHANGES_REQUESTED** [review]
Thanks, just a minor comment

**Comment by childrentime** [comment]
Done！

**Review by Tyriar — APPROVED** [review]

**Pull request merged**